ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/preexecution/PreExecutingSubmissionValidatorSpec.scala

```diff
@@ -34,10 +34,45 @@ import org.scalatest.wordspec.AsyncWordSpec
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
-class PreExecutingSubmissionValidatorSpec extends AsyncWordSpec with Matchers with MockitoSugar {
+class PreExecutingSubmissionValidatorSpec
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
   private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
 
   "validate" should {
+
+    "read from ledger state only once in presence of (KV) state key inputs of DAML contract keys'" in {
```

Comment:
```suggestion
    "read from ledger state only once in presence of (KV) state key inputs of DAML contract keys'" in {
```
```suggestion
    "read from ledger state only once even if a DAML contract key is in the read-set" in {
```

Reply:
Done but `even` -> `when` because I'm not testing all the cases.